src/java/org/apache/cassandra/service/accord/AccordSafeCommand.java

```diff
@@ -127,9 +127,11 @@ public Command original()
         return original;
     }
 
-    public SavedCommand.SavedDiff diff()
+    public SavedCommand.Writer<TxnId> diff()
     {
-        return SavedCommand.diff(original, current);
+        if (original == current || current == null)
```

Comment:
when we don't know the txn we don't have a `null` but a `Uninitialised`, this causes us to do more work than needed as we still allocate the diff.

Fixed and tested in https://github.com/ifesdjeen/cassandra/pull/2